polkadot/zombienet-sdk-tests/src/lib.rs

```diff
@@ -0,0 +1,226 @@
+use serde_json::json;
+use subxt::{OnlineClient, PolkadotConfig};
+use subxt_signer::sr25519::dev;
+use zombienet_sdk::{
+	LocalFileSystem, Network, NetworkConfigBuilder, NetworkConfigExt, NetworkNode,
+	RegistrationStrategy,
+};
+
+#[cfg(test)]
+mod tests;
+
+#[subxt::subxt(runtime_metadata_path = "artifacts/polkadot_metadata_small.scale")]
+pub mod polkadot {}
+
+pub type Error = Box<dyn std::error::Error>;
+
+pub fn runtime_config() -> serde_json::Value {
+	json!({
+		"configuration": {
+			"config": {
+				"max_validators_per_core": 1,
+				"needed_approvals": 1,
+				"group_rotation_frequency": 10
+			}
+		}
+	})
+}
+
+/// Required bins: polkadot, malus, polkadot-parachain
+/// built with `--features fast-runtime`
+///
+/// This spawns a network with 3 honest validators and 1 malicious backer.
+/// One parachain with id 2000 and one cumulus-based collator.
+pub async fn spawn_network_malus_backer() -> Result<Network<LocalFileSystem>, Error> {
+	let network = NetworkConfigBuilder::new()
+		.with_relaychain(|r| {
+			let patch = runtime_config();
+
+			r.with_chain("westend-local")
+				.with_genesis_overrides(patch)
+				.with_default_command("polkadot")
+				.with_default_args(vec![
+					"--no-hardware-benchmarks".into(),
+					"--insecure-validator-i-know-what-i-do".into(),
+					"-lparachain=debug".into(),
+				])
+				.with_node(|node| node.with_name("honest-0"))
+				.with_node(|node| node.with_name("honest-1"))
+				.with_node(|node| node.with_name("honest-2"))
+				.with_node(|node| {
+					node.with_name("malicious-backer")
+						.with_command("malus")
+						.with_subcommand("suggest-garbage-candidate")
+						.with_args(vec![
+							"--no-hardware-benchmarks".into(),
+							"--insecure-validator-i-know-what-i-do".into(),
+							"-lMALUS=trace".into(),
+						])
+				})
+		})
+		.with_parachain(|p| {
+			p.with_id(2000)
+				.cumulus_based(true)
+				.with_registration_strategy(RegistrationStrategy::InGenesis)
+				.with_collator(|n| n.with_name("collator").with_command("polkadot-parachain"))
+		})
+		.build()
+		.unwrap()
+		.spawn_native()
```

Comment:
Yes, this should be `k8s`. I think we can reuse the images from the current zombienet tests (https://github.com/paritytech/polkadot-sdk/blob/5ba8921787609dddbabb8be3443186b65fdde190/.gitlab/pipeline/zombienet/polkadot.yml). 

Any chance you have got a gitlab pipeline config which I can adapt for these tests? :innocent:

Reply:
Yes, the integration already works from gitlab pipelines. Did you want to trigger this manually? Also, we should define the image versions to use, since we currently use the image created by each `pr`.
Thanks!

Reply:
Nice! Image created for each PR is the right thing to do, but the PR should be rebased ontop of https://github.com/paritytech/polkadot-sdk/pull/2226 (I'll handle this).

How can we 'port` the 'path hack' to k8s? I speak about [this](https://github.com/paritytech/polkadot-sdk/pull/3442/files#diff-c13ffd925dbec3ff1c97780870e9e489bacbe07fe77e39b2d4438dfa2f9d54caR244).

Reply:
For the `path hack`, where we get the new binaries? are the one built by the pr? In a similar test we use an image from `master` as base process and then we restart the node with the binary download from the pr.

https://github.com/paritytech/polkadot-sdk/blob/master/polkadot/zombienet_tests/misc/0002-upgrade-node.zndsl#L8-L21

Thx!

Reply:
> For the path hack, where we get the new binaries?
The new binaries are the ones from the PR. The old ones - some past release. Let's say polkadot 1.7. The runtime test in particular will be deprecated at some point. It won't be needed once the runtime is upgraded everywhere.

Reply:
Sounds good, so I will modify the test to start the network with polkadot v1.7 and upgrade the binary with the ones built in the pipeline for the pr.